source/Lib/DecoderLib/DecCu.cpp

```diff
@@ -910,4 +951,73 @@ void DecCu::xDeriveCUMV( CodingUnit &cu, MotionHist& hist )
   }
 }
 
+PelUnitBuf DecCu::getPredBuf(const CodingUnit &unit)
+{
+  PelUnitBuf ret;
+  ret.chromaFormat = unit.chromaFormat;
+  ret.bufs.resize_noinit( getNumberValidComponents( unit.chromaFormat ) );
+
+  if( unit.Y().valid() )
+  {
+    ret.bufs[0].buf    = m_predBuf.bufs[0].buf;
+    ret.bufs[0].stride = unit.blocks[0].width;
+    ret.bufs[0].width  = unit.blocks[0].width;
+    ret.bufs[0].height = unit.blocks[0].height;
+  }
+
+  if( isChromaEnabled( unit.chromaFormat ) )
+  {
+    if( unit.Cb().valid() )
+    {
+      ret.bufs[1].buf    = m_predBuf.bufs[1].buf;
+      ret.bufs[1].stride = unit.blocks[1].width;
+      ret.bufs[1].width  = unit.blocks[1].width;
+      ret.bufs[1].height = unit.blocks[1].height;
+    }
+
+    if( unit.Cr().valid() )
+    {
+      ret.bufs[2].buf    = m_predBuf.bufs[2].buf;
+      ret.bufs[2].stride = unit.blocks[2].width;
+      ret.bufs[2].width  = unit.blocks[2].width;
+      ret.bufs[2].height = unit.blocks[2].height;
+    }
+  }
+
+  return ret;
+}
+
+const CPelUnitBuf DecCu::getPredBuf(const CodingUnit &unit) const
+{
+  CPelUnitBuf ret;
+  ret.chromaFormat = unit.chromaFormat;
+  ret.bufs.resize( 3 );
```

Comment:
Consider using `resize_noinit` with proper number of components, similar to the non-const variant. All the stuff will be manually initalized after.